aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -357,6 +388,18 @@ public class AwsProperties implements Serializable {
    */
   public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
 
+  /**
+   * Used by {@link GlueCatalog}.
+   * The table name is used as part of table specific catalog properties.
+   */
+  public static final String TABLE_NAME = "table-name";
```

Comment:
1. The `table-name` and `db-name` properties are quite hacky... why not just dynamically add the tags based on catalog tag prefix `s3.write.tags.`? For example, if table name tag enabled, add property `s3.write.tags.table-name` in Glue catalog.
2. the tag names themselves, I think we should probably prefix them by `iceberg.`, for example `iceberg.table-name`, `iceberg.namespace-name`.
3. the variable name should really indicate this is a special tag name, for example `S3_TABLE_SPECIFIC_WRITE_TAGS_TABLE_NAME`.

Reply:
I have made the changes.